pkg/controller/user-cluster-controller-manager/resources/resources/core-dns/networkpolicy.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package coredns
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"k8c.io/kubermatic/v2/pkg/controller/operator/common"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+)
+
+// AllowAllDnsNetworkPolicyCreator allows egress traffic from all pods to kube-dns
+func AllowAllDnsNetworkPolicyCreator() reconciling.NamedNetworkPolicyCreatorGetter {
```

Comment:
I think this policy selects all the pods from all the namespaces and allows traffic only to DNS. 
Wouldn't that block traffic between the pods?

Reply:
https://kubernetes.io/docs/concepts/services-networking/network-policies/#isolated-and-non-isolated-pods 

> Pods become isolated by having a NetworkPolicy that selects them. Once there is any NetworkPolicy in a namespace selecting a particular pod, that pod will reject any connections that are not allowed by any NetworkPolicy. (Other pods in the namespace that are not selected by any NetworkPolicy will continue to accept all traffic.)

This is what it looks like on my cluster. 

```
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  creationTimestamp: "2021-12-02T19:01:39Z"
  generation: 1
  name: allow-dns
  namespace: kube-system
  resourceVersion: "128709"
  uid: e009eb81-5f2a-4524-a576-5e1b7423d162
spec:
  egress:
    - ports:
        - port: 53
          protocol: TCP
        - port: 53
          protocol: UDP
      to:
        - podSelector:
            matchLabels:
              app.kubernetes.io/name: kube-dns
    - ports:
        - port: 53
          protocol: TCP
        - port: 53
          protocol: UDP
      to:
        - ipBlock:
            cidr: 169.254.20.10/32
  podSelector: {}
  policyTypes:
    - Ingress
    - Egress
```

Reply:
Yes it block traffic between the pods, thats the point of the network policy. The name was changed however to clarify that this is the default policy